### ISSUE TITLE
HTML entrypoint with script refering to package not working

### ISSUE THREAD

**Issue by simonihmig** [embroider-build/embroider#1305]
According to https://github.com/simonihmig/ember-responsive-image/issues/412#issuecomment-1176574070, our `index.html` serving as webpack's entrypoint should allow referring to npm packages as module-type scripts, as in

```html
<script src="@ember-responsive-image/loader/blurhash-script" type="module"></script>
```

However this throws with `Error: Can't resolve './@ember-responsive-imageloader/blurhash-script' in '$TMPDIR/embroider/197651/test-app'`. Note two things here:
* the `./` prefix is obviously wrong 
👉 [Source](https://github.com/embroider-build/embroider/blob/main/packages/webpack/src/ember-webpack.ts#L191)
* the first found dash is removed (`@ember-responsive-imageloader` should be `@ember-responsive-image/loader`). 
👉 [Source](https://github.com/embroider-build/embroider/blob/main/packages/core/src/html-entrypoint.ts#L54)

*Preparing a PR...*

**Comment by ef4** [comment]
The thing I said in that comment is true about how the core of embroider itself works, but the compatibility system is not exposing those features directly to users right now, because `app/index.html` has a bunch of classic behavior. Right now, references in index.html refer to the build output, not the input. Which is wrong, but how ember has always done it. We reverse engineer from those what you really want your inputs to be.

I think we need to introduce a flag that opts people in new-style HTML files that would let them do this sort of thing without the compatibility layer getting in the way.

**Comment by simonihmig** [comment]
What do you mean by "input" here, the actual source, or the *output* of stage 2, before bundling?

Have you seen what I did in #1306? It does fix the issue for me, but it might still not be the right thing to do!? 🤔 

I think it basically treats any absolute path (starting with `/` or whatever `rootUrl` is) as *relative* to the stage 2 output (`pathToVanillaApp`), while anything else would get resolved according to node module resolution.

**Comment by ef4** [comment]
> What do you mean by "input" here, the actual source, or the output of stage 2, before bundling?

I think the actual source, although I could be convinced that might in practice mean the output of stage 2. Really I'm hoping the output of stage2 and the actual source will begin to converge, so that we rely less and less on the things (like app tree merging) that go in stage 2.

> Have you seen what I did in https://github.com/embroider-build/embroider/pull/1306? It does fix the issue for me, but it might still not be the right thing to do!? 🤔

I did see it, and I'm not sure yet.

The thing I'm worried about is that right now I'm pretty sure `<script src="some/file.js"></script>` works and will cause `public/some/file.js` to execute. I think this change would alter that to look for an NPM package named `some`?

In other words, we need to commit to either saying that a path without a `/` or `.` prefix is an npm-resolvable path or a relative URL but it can't be both, and I think the classic behavior is that it's a relative URL.

**Comment by simonihmig** [comment]
> The thing I'm worried about is that right now I'm pretty sure <script src="some/file.js"></script> works and will cause public/some/file.js to execute. I think this change would alter that to look for an NPM package named some?

Yeah, right. In practice for the apps that I worked on these were always with a leading `/`, but when that is not the case the change would be breaking I think. That's why I was unsure about it as well...

Could we make thing depend on `type="module"` instead? Which was not really a thing in classes Ember builds, right?
Like use node module resolution for any `<script type="module">` entrypoint, and keep things as-is for everything else. Does that make sense?
